Source/SwiftLintFramework/Rules/IdentifierNameRule.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  IdentifierNameRule.swift
+//  SwiftLint
+//
+//  Created by JP Simard on 5/16/15.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct IdentifierNameRule: ASTRule, ConfigurationProviderRule {
+
+    public var configuration = NameConfiguration(minLengthWarning: 3,
+                                                 minLengthError: 2,
+                                                 maxLengthWarning: 40,
+                                                 maxLengthError: 60)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "identifier_name",
+        name: "Identifier Name",
+        description: "Identifier names should only contain alphanumeric characters and " +
+            "start with a lowercase character or should only contain capital letters. " +
+            "In an exception to the above, variable names may start with a capital letter " +
+            "when they are declared static and immutable. Variable names should not be too " +
+            "long or too short.",
+        nonTriggeringExamples: [
+            "let myLet = 0",
+            "var myVar = 0",
+            "private let _myLet = 0",
+            "class Abc { static let MyLet = 0 }",
+            "let URL: NSURL? = nil",
+            "let XMLString: String? = nil",
+            "enum Foo { case myEnum }",
+            "func isOperator(name: String) -> Bool",
+            "func typeForKind(_ kind: SwiftDeclarationKind) -> String",
+            "func == (lhs: SyntaxToken, rhs: SyntaxToken) -> Bool"
+        ],
+        triggeringExamples: [
+            "↓let MyLet = 0",
+            "↓let _myLet = 0",
+            "private ↓let myLet_ = 0",
+            "↓let myExtremelyVeryVeryVeryVeryVeryVeryLongLet = 0",
+            "↓var myExtremelyVeryVeryVeryVeryVeryVeryLongVar = 0",
+            "private ↓let _myExtremelyVeryVeryVeryVeryVeryVeryLongLet = 0",
+            "↓let i = 0",
+            "↓var id = 0",
+            "private ↓let _i = 0",
+            "enum Foo { case ↓MyEnum }",
+            "↓func IsOperator(name: String) -> Bool"
+        ]
+    )
+
+    private func nameIsViolatingCase(_ name: String) -> Bool {
+        let secondIndex = name.characters.index(after: name.startIndex)
+        let firstCharacter = name.substring(to: secondIndex)
+        if firstCharacter.isUppercase() {
```

Comment:
This might benefit from using `guard`s:

```diff
-if firstCharacter.isUppercase() {
-    if name.characters.count > 1 {
-        let range = secondIndex..<name.characters.index(after: secondIndex)
-        let secondCharacter = name.substring(with: range)
-        return secondCharacter.isLowercase()
-    }
-    return true
-}
-return false
+guard firstCharacter.isUppercase() else { return false }
+guard name.characters.count > 1 else { return true }
+let range = secondIndex..<name.characters.index(after: secondIndex)
+let secondCharacter = name.substring(with: range)
+return secondCharacter.isLowercase()
```